### DISCUSSION TITLE
Allow for specifying UUID attr in CREATE?

### DISCUSSION THREAD

**Discussion by dvv** [kanidm/kanidm#2552] [Q&A] [answered]
Hi!

I wonder if we can go with the subj? The rationale is to re-use external system identifier (e.g., database id)  -- it can ease binding between systems.

TIA

**Comment by Firstyear** [comment]
What interface do you intend to use here for this? We have a SCIM bulk import for sync, or do you mean the normal rest api?

**Reply by dvv** [reply]
I couldn't find a curl-ish example of SCIM import -- is it a one-step operation?
So I think it could be enabled for the normal rest api.

**Reply by Firstyear** [reply]
It's part of the ipa/ldap sync tools. What are you trying to achieve specifically ?You may want a "migration tool" like we have that handles a lot of this for you.

**Reply by dvv** [reply]
Tool won't go in the case I'm exploring.
I'd like to create person/group from a after-insert db trigger. Can you point me out what to tweak in the code _locally_ so that it would allow me for specifyng `"uuid": [...]` in attrs for POST?

**Reply by Firstyear** [reply]
But you can use the tool as a structure to guide your own impl if you want one.

Otherwise, in the post, just specify `uuid: "..."` which should work if you are posting to athe rest interface.

**Reply by dvv** [reply]
Posting
```
{"attrs":{
"name": ["test.post"],
"uuid": ["7661c296-b826-410e-8c88-2dd42a8de852"]
}}
```
gives `create: {"class", "name", "uuid"} !⊆ allowed: {"account_expire", "account_valid_from", "class", "displayname", "mail", "name"}`

**Reply by Firstyear** [reply]
Which account are you posting as? That's an access control issue, not anything else.

**Reply by dvv** [reply]
As a service-account which shows the following:
```
memberof: idm_people_pii_read
memberof: idm_schema_admins
memberof: idm_people_admins
memberof: idm_group_admins
memberof: idm_oauth2_admins
memberof: idm_all_accounts
memberof: idm_account_mail_read
memberof: idm_people_on_boarding
memberof: idm_high_privilege
```

**Reply by Firstyear** [reply]
What api end point are you posting to?

**Reply by dvv** [reply]
`/v1/person`

**Reply by yaleman** [reply]
I don't think the ACPs actually allow writing to the UUID attribute, just had a quick look and groups can *search* but can't find any mod permissions...

**Reply by dvv** [reply]
Right. Does allowing for specifying "uuid" mount well with our vision on security? If yes I would like to have this able. If no I will just be looking for workarounds.

**Reply by Firstyear** [reply, marked as answer]
@dvv writing uuids is fine. I think we just need it as an extended option for people onboarding.
